assets/js/modules/analytics/datastore/report.js

```diff
@@ -0,0 +1,151 @@
+/**
+ * modules/analytics data store: report.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+import isPlainObject from 'lodash/isPlainObject';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import Data from 'googlesitekit-data';
+import { STORE_NAME } from './constants';
+import { stringifyObject } from '../../../util';
+import { createFetchStore } from '../../../googlesitekit/data/create-fetch-store';
+import {
+	isValidMetrics,
+	isValidDimensions,
+	isValidDateRange,
+	isValidOrders,
+} from '../../../util/report-validation';
+
+const fetchGetReportStore = createFetchStore( {
+	baseName: 'getReport',
+	controlCallback: ( { options } ) => {
+		return API.get( 'modules', 'analytics', 'report', options );
+	},
+	reducerCallback: ( state, report, { options } ) => {
+		return {
+			...state,
+			reports: {
+				...state.reports,
+				[ stringifyObject( options ) ]: report,
+			},
+		};
+	},
+	argsToParams: ( options ) => {
+		const {
+			metrics,
+			dimensions,
+			orderby,
+		} = options;
+
+		invariant( isPlainObject( options ), 'Options for Analytics report must be an object.' );
+		invariant( isValidDateRange( options ), 'Either date range or start/end dates must be provided for Analytics report.' );
+
+		if ( metrics ) {
+			invariant(
+				isValidMetrics( metrics ),
+				'Metrics for an Analytics report must be either a string, an array of strings, an object, an array of objects or a mix of strings and objects. If an object is used, it must have "expression" and "alias" properties.',
+			);
+		}
```

Comment:
Since `metrics` is required, this `invariant` should always run, not just if `metrics` is available.

Reply:
Updated.